Save data every couple of seconds

bbrk24 opened the pull request:
For battery testing, I will make a small local change to circumvent the need for a radio command to start it:
```diff
-    if (radioUart.available()) {
+    if (true) {
       //The string returned from readStringUntil does not include the terminator
-      String command = radioUart.readStringUntil('\n');
+      String command = "start";
```

bbrk24 committed:
Save data every couple of seconds

CClarke513 reviewed (dismissed):
Fine by me!

bbrk24 commented:
Marking this as draft until I figure out the performance issue

DarthSethious1 commented on sketch_oct9a.ino:
```diff
@@ -324,6 +343,7 @@ void setup() {
 
   Scheduler.startLoop(gps_loop);
   Scheduler.startLoop(radio_loop);
+  Scheduler.startLoop(save_loop);
```
I understand this is necessary, but I feel like we should closely watch/look at how many different loops we're starting on a single-core system. If we put the 'saveToSD' stuff in the save loop as another, would it be quicker?

bbrk24 replied:
Yes, I was able to make it a bit faster by merging it with the GPS loop, but still nowhere near as fast as it was before. It's possible that simply writing the data to the SD card is a bottleneck and we should be careful how/when we do it.

DarthSethious1 replied:
I think you're looking into that already, so keep messing around with it and try and find what's best

DarthSethious1 commented on sketch_oct9a.ino:
```diff
@@ -374,4 +379,12 @@ void radio_loop() {
 
+void save_loop() {
+  delay(2000);
+  // If the file was intentionally closed, don't re-open it.
```
good save

DarthSethious1 reviewed (approved):
I'd like my comments to at least be commented on before I merge, but this looks good

bbrk24 committed:
Fix performance issues

Besides moving the definition of GPS_FREQ (which I did for code
organization), and setting ledsOn to false (which is necessary for
correctness, but I missed earlier), every change in this commit is
proven to improve the performance of the code.

CClarke513 reviewed (approved):
All good by me - Riley got things down to less than 1 Hz difference

DarthSethious1 reviewed (approved):
Everything looked good to me. Awesome job at ID-ing the problem and fixing it all within like 36 hours. Both you, Riley, and Cam are doing very well.
